client/src/main/java/io/split/engine/matchers/DependencyMatcher.java

```diff
@@ -0,0 +1,71 @@
+package io.split.engine.matchers;
+
+import io.split.client.SplitClientImpl;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Supports the logic: if user is in split "feature" treatments ["on","off"]
+ */
+public class DependencyMatcher implements Matcher {
+    private String _split;
+    private List<String> _treatments;
+
+    public DependencyMatcher(String split, List<String> treatments) {
+        _split = split;
+        _treatments = treatments;
+    }
+
+    @Override
+    public boolean match(Object matchValue, String bucketingKey, Map<String, Object> attributes, SplitClientImpl splitClient) {
+        if (matchValue == null) {
+            return false;
+        }
+
+        if (!(matchValue instanceof String)) {
+            return false;
+        }
+
+        String result = splitClient.getTreatmentWithoutImpressions(
+                (String) matchValue,
+                bucketingKey,
+                _split,
+                attributes
+        );
+
+//        if(Treatments.isControl(result)) {
```

Comment:
I personally believe that we should return control if the parent does, and implemented accordingly. We made a call this morning to not do that, but I figured for now I'd comment out the code rather than fully ripping it out in case we change back.